llama-index-core/llama_index/core/objects/base.py

```diff
@@ -113,21 +169,41 @@ def from_objects(
         index_cls: Type[BaseIndex] = VectorStoreIndex,
         **index_kwargs: Any,
     ) -> "ObjectIndex":
+        # pick the best mapping if not provided
         if object_mapping is None:
-            object_mapping = SimpleObjectNodeMapping.from_objects(objects)
+            object_mapping = cls.get_object_mapping(objects)
+
         nodes = object_mapping.to_nodes(objects)
         index = index_cls(nodes, **index_kwargs)
         return cls(index, object_mapping)
 
+    @classmethod
+    def from_objects_and_index(
```

Comment:
is this used? 

actually nvm i see why its useful (e.g. connecting to an existing index). is this how we'd save an index with a vector store?

Reply:
Yea, I added an example of using chroma and this method to the object index notebook